### ISSUE TITLE
Partition size not enforced

### ISSUE THREAD

**Issue by trevorkarn** [sagemath/sage#35579]
### Is there an existing issue for this?

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.

### Did you read the documentation and troubleshoot guide?

- [X] I have read the documentation and troubleshoot guide

### Environment

```markdown
- **OS**: macOS Ventura 13.3.1
- **Sage Version**: 10.0.beta0
```

### Steps To Reproduce

We require a size to be specified when creating a partition iterator:
```
sage: Partitions(starting=[2,1,1])
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In [7], line 1
----> 1 Partitions(starting=[Integer(2),Integer(1),Integer(1)])

File ~/Applications/sage/src/sage/misc/classcall_metaclass.pyx:320, in sage.misc.classcall_metaclass.ClasscallMetaclass.__call__()
    318 """
    319 if cls.classcall is not None:
--> 320     return cls.classcall(cls, *args, **kwds)
    321 else:
    322     # Fast version of type.__call__(cls, *args, **kwds)

File ~/Applications/sage/src/sage/combinat/partition.py:5886, in Partitions.__classcall_private__(cls, n, **kwargs)
   5884                 if 'max_length' in kwargs:
   5885                     return RegularPartitions_truncated(kwargs['regular'], kwargs['max_length'])
-> 5886         raise ValueError("the size must be specified with any keyword argument")
   5887     return Partitions_all()
   5889 raise ValueError("n must be an integer or be equal to one of "
   5890                  "None, NN, NonNegativeIntegers()")

ValueError: the size must be specified with any keyword argument
```
but it is not used
```
sage: for p in Partitions(2, starting=[3,2,1]): print(p)
[3, 2, 1]
[3, 1, 1, 1]
[2, 2, 2]
[2, 2, 1, 1]
[2, 1, 1, 1, 1]
[1, 1, 1, 1, 1, 1]
```

### Expected Behavior

If the starting partition is not a partition of the required size, then an error should be thrown.

### Actual Behavior

See "Steps to reproduce".

**Comment by DaveWitteMorris** [comment]
This issue should be closed, because it was fixed by [pull request #35582](https://github.com/sagemath/sage/pull/35582).

**Comment by trevorkarn** [comment]
Thanks!
